askama_derive/src/generator.rs

```diff
@@ -1008,10 +1073,36 @@ impl<'a> Generator<'a> {
         self.buf_writable.push(Writable::Expr(s));
     }
 
-    // Write expression buffer and empty
-    fn write_buf_writable(&mut self, buf: &mut Buffer) -> Result<usize, CompileError> {
+    fn generate_expr_id(
+        &mut self,
+        buf_expr: &mut Buffer,
+        is_cacheable: bool,
+        expression: String,
+        expr_cache: &mut HashMap<String, usize>,
+    ) -> Result<usize, CompileError> {
+        match expr_cache.entry(expression) {
+            Entry::Occupied(e) if is_cacheable => Ok(*e.get()),
+            e => {
+                let id = self.named;
+                self.named += 1;
+
+                buf_expr.write(&format!("expr{id} = "));
+                buf_expr.write("&");
+                buf_expr.write(e.key());
+                buf_expr.writeln(",")?;
+
+                if let Entry::Vacant(e) = e {
+                    e.insert(id);
+                }
+
+                Ok(id)
+            }
+        }
+    }
+
+    fn buf_writable(&mut self, indent: u8) -> Result<WriteParts, CompileError> {
```

Comment:
As for the name -- can you describe what this does now?

Reply:
It generates the literal string and the format arguments without adding the "wrapper" (ie `::std::write!(` or `format!(`).

Reply:
Maybe `prepare_format()` or some such?